hs/src/Reach/Backend/JS.hs

```diff
@@ -456,19 +456,20 @@ jsExpr = \case
             [ ("amt", zero')
             , ("tok", tok')
             ]
-  DLE_CanReceive _ addr tok -> do
+  DLE_TokenAccepted _ acc tok -> do
     (ctxt_mode <$> ask) >>= \case
       JM_Backend -> do
-        -- TODO - I'm not really sure what to do here, but I imagine I should do something similar to the simulate case.
-        --addr' <- jsArg addr
-        --tok' <- jsArg tok
-        --return $ "await" <+> jsApply (addr' <> ".tokenAccepted") [tok']
-        return "false"
+        -- TODO - This is wrong, but I'm still not sure what to do here.
+        return $ "false"
       JM_View -> impossible "view canReceive"
       JM_Simulate -> do
-        addr' <- jsArg addr
+        acc' <- jsArg acc
         tok' <- jsArg tok
-        return $ "await" <+> jsApply (addr' <> ".tokenAccepted") [tok']
```

Comment:
You should do this part AND add the `tokenAccepted` sim txn

Reply:
The code you have is the wrong type, because you're returning a sim txn where a bool is expected

Reply:
Help me understand JS.hs a little more.  The `JM_Simulate` code I'm writing only runs for Algorand as far as I can tell.  Also it's wrong, because I have the address as a string and I'm not sure how to convert it into an address to do `acc.tokenAccepted(tok)` (and crashing there for ALGO but not for ETH is how I know it's only running for Algorand).  Under what conditions are things simulated, and why do I want to call `.tokenAccepted` in the simulate case but not the backend case?

Reply:
- Yes, it only runs for Algorand
- Look under the covers of `acc.tokenAccepted(tok)` and call a helper function (that maybe you need to make) called `accountTokenAccepted(acc, tok)` that doesn't need an account object
- On Algorand, we have to simulate before EVERY consensus step to learn what resources are used. That's why we have the sim txn to learn that the account and token are used. But, suppose the program is `transfer(1).to(tokenAccepted(a, b) ? c : d)`. We need to decide whether to include `c` or `d`.
- In the backend case, because you did the emit log, the value will be totally ignored anyways... we'll look at the log and find out what the real consensus did when it ran (because the "backend" code runs AFTER the consensus step, whereas the "simulate" code runs BEFORE)